proposals/readonly-methods.md

```diff
@@ -0,0 +1,65 @@
+# Readonly Methods
+
+* [x] Proposed
+* [ ] Prototype: Not Started
+* [ ] Implementation: Not Started
+* [ ] Specification: Not Started
+
+## Summary
+[summary]: #summary
+
+Provide a way to specify individual methods on a struct do not modify state, in the same way that `readonly struct` specifies no instance method modifies state.
+
+It is worth noting that `readonly method` != `pure method`. A `pure` method guarantees no state will be modified. A `readonly` method only guarantees that instance state will not be modified.
+
+All instance methods on a `readonly struct` could be considered implicitly `readonly methods`. Explicit `readonly methods` declared on non-readonly structs would behave in the same manner. For example, they would still create hidden copies if you called an instance method (on the current instance or on a field of the instance) which was itself not-readonly.
+
+## Motivation
+[motivation]: #motivation
```

Comment:
Why do you add this?

Reply:
It is in the template, and is what allows you to link to sub-sections of the markdown file.